### PR TITLE
Pins debugging improvements

### PR DESCRIPTION
**Pull request by InsanityAutomation**
Add ability to explicitly set pin mode, attempt to restore prior mode on M43 toggle

### CODE REVIEW

File: Marlin/src/gcode/control/M42.cpp [hunk 1]
```diff
@@ -37,16 +37,33 @@
...
+    const uint8_t pin_mode = parser.value_byte();
+    if (pin_mode == 0)
+      pinMode(pin, INPUT);
+    else if (pin_mode == 1)
+      pinMode(pin, OUTPUT);
+    else if (pin_mode == 2)
+      pinMode(pin, INPUT_PULLUP);
+    else
+      SERIAL_ECHOLNPGM("Invalid Pin Mode");
+    return;
```
Comment (thinkyhead) [inline Marlin/src/gcode/control/M42.cpp:61]: Is this extension to `M42` needed for all architectures including the tiny ones, or can we make it dependent on `PINS_DEBUGGING`?

Comment (InsanityAutomation) [reply Marlin/src/gcode/control/M42.cpp:61]: I don't have an issue with all of m42 requiring pins debugging personally.

Comment (thinkyhead) [reply Marlin/src/gcode/control/M42.cpp:61]: BTAIM, the code `M42` is a lightweight and longstanding feature that simply allows setting a pin for those cases where someone wants to add a simple device to a machine but doesn't want to (or can't) install a new firmware. I added the `M43` code much later, and it's a very costly debug feature. So we wouldn't want to make `M42` require `M43`.

**Comment by thinkyhead** [comment]
When extending a G-code don't forget to add the new parameters to the RepRap wiki. This change is all set in the documentation.

**Pull request merged**